Assets/MixedRealityToolkit/Utilities/MixedRealityServiceRegistry.cs

```diff
@@ -0,0 +1,231 @@
+        /// </summary>
+        /// <param name="interfaceType">The interface type of the service being removed.</param>
+        /// <param name="serviceInstance">Instance of the service to remove.</param>
+        /// <param name="registrar">Instance of the registrar manages the service.</param>
+        /// <returns>
+        /// True if the service was successfully removed, false otherwise.
+        /// </returns>
+        private static bool RemoveServiceInternal(
+            Type interfaceType,
+            IMixedRealityService serviceInstance,
+            IMixedRealityServiceRegistrar registrar)
+        {
+            if (!registry.ContainsKey(interfaceType)) { return false; }
+
+            List<KeyValuePair<IMixedRealityService, IMixedRealityServiceRegistrar>> services = registry[interfaceType];
+
+            return services.Remove(new KeyValuePair<IMixedRealityService, IMixedRealityServiceRegistrar>(serviceInstance, registrar));
+        }
+
+        /// <summary>
+        /// Gets the instance of the requested service from the registry.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service being requested.</typeparam>
+        /// <param name="serviceInstance">Output parameter to receive the requested service instance.</param>
+        /// <param name="registrar">Output parameter to receive the registrar that loaded the service instance.</param>
+        /// <param name="name">Optional name of the service.</param>
+        /// <returns>
+        /// True if the requested service is being returned, false otherwise.
+        /// </returns>
+        public static bool TryGetService<T>(
+            out T serviceInstance,
+            out IMixedRealityServiceRegistrar registrar,
+            string name = null)
+        {
+            Type interfaceType = typeof(T);
+
+            if (!registry.ContainsKey(interfaceType))
+            {
+                serviceInstance = default(T);
+                registrar = null;
+                return false;
+            }
+
+            List<KeyValuePair<IMixedRealityService, IMixedRealityServiceRegistrar>> services = registry[interfaceType];
+            
+            int registryIndex = -1;
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                // Find the desired service by it's name.
+                for (int i = 0; i < services.Count; i++)
+                {
+                    if (services[i].Key.Name != name) { continue; }
+
+                    registryIndex = i;
+                    break;
+                }
+
+                if (registryIndex == -1)
+                {
+                    // Failed to find the requested service.
+                    serviceInstance = default(T);
+                    registrar = null;
+                    return false;
+                }
+            }
+            else
+            {
+                // Use the first service found
+                registryIndex = 0;
```

Comment:
>registryIndex = 0; [](start = 16, length = 18)

Can you still add a Debug.Assert or warn or log or return failure if there are more than two services? This will be an undefined thing that could really bite people in the future (i.e. things could 'just work' luckily because it happens to be that the first dude always asks but the second dude doesn't, but if the second dude ever asks it's gonna be bad)